obspy/clients/syngine/tests/test_client.py

```diff
@@ -0,0 +1,489 @@
+                        unicode_literals)
+from future.builtins import *  # NOQA @UnusedWildImport
+from future.utils import native_str
+
+import io
+import unittest
+
+import numpy as np
+
+import obspy
+from obspy.core.compatibility import mock
+from obspy.core.util.base import NamedTemporaryFile
+from obspy.clients.syngine import Client
+from obspy.clients.base import DEFAULT_TESTING_USER_AGENT, ClientHTTPException
+
+BASE_URL = "http://service.iris.edu/irisws/syngine/1"
+
+
+class RequestsMockResponse(object):
+    def __init__(self):
+        self.text = ""
+        self.content = b""
+        self.status_code = 200
+        self._json = {}
+
+    def json(self):
+        return self._json
+
+
+class ClientTestCase(unittest.TestCase):
+    """
+    Test cases for obspy.clients.iris.client.Client.
+    """
+    c = Client(user_agent=DEFAULT_TESTING_USER_AGENT)
+
+    def test_get_model_info_mock(self):
+        """
+        Mock test for the get_model_info() method.
+        """
+        with mock.patch("requests.get") as p:
+            r = RequestsMockResponse()
+            r._json["slip"] = [0.0, 1.0, 2.0]
+            r._json["sliprate"] = [0.0, 1.0, 2.0]
+            p.return_value = r
+            self.c.get_model_info("test_model")
+
+        self.assertEqual(p.call_count, 1)
+        self.assertEqual(p.call_args[1]["url"],
+                         'http://service.iris.edu/irisws/syngine/1/info')
+        self.assertEqual(p.call_args[1]["params"],
+                         {'model': 'test_model'})
+        self.assertEqual(p.call_args[1]["headers"],
+                         {'User-Agent': DEFAULT_TESTING_USER_AGENT})
+
+    def test_get_model_info(self):
+        """
+        Actual test for the get_model_info() method.
+        """
+        info = self.c.get_model_info("test")
+
+        self.assertTrue(isinstance(info, obspy.core.AttribDict))
+        # Check two random keys.
+        self.assertEqual(info.dump_type, "displ_only")
+        self.assertEqual(info.time_scheme, "newmark2")
+        # Check that both arrays have been converted to numpy arrays.
+        self.assertTrue(isinstance(info.slip, np.ndarray))
+        self.assertTrue(isinstance(info.sliprate, np.ndarray))
+
+    def test_get_available_models_mock(self):
+        with mock.patch("requests.get") as p:
+            p.return_value = RequestsMockResponse()
+            self.c.get_available_models()
+
+        self.assertEqual(p.call_count, 1)
+        self.assertEqual(p.call_args[1]["url"],
+                         'http://service.iris.edu/irisws/syngine/1/models')
+        self.assertEqual(p.call_args[1]["params"], None)
+        self.assertEqual(p.call_args[1]["headers"],
+                         {'User-Agent': DEFAULT_TESTING_USER_AGENT})
+
+    def test_get_available_models(self):
+        models = self.c.get_available_models()
+        self.assertTrue(isinstance(models, dict))
+        keys = list(models.keys())
```

Comment:
I'm not sure how `future` handles it, but I don't think you actually need to convert this into a list.

However, you don't need to get the keys: `len(list(models.keys()))` -> `len(models.keys())` -> `len(models)` and `'name' in list(models.keys())` -> `'name' in models.keys()` -> `'name' in models`.